### ISSUE TITLE
Under specific circumstances, it may cause a dead cycle

### ISSUE THREAD

**Issue by 549915217** [josdejong/mathjs#3020]
```javascript
 while (!hasOwnProperty.call(queue, ++base)) continue
```
The `hasOwnProperty` method may be polluted in the current environment, leading to incorrect scoping and causing an infinite loop. It is recommended to use the more reliable syntax `Object.prototype.hasOwnProperty.call` instead.The file path is `src\utils\lruQueue.js`

Here is the updated code:

```javascript
while (!Object.prototype.hasOwnProperty.call(queue, ++base)) continue;
```

By using `Object.prototype.hasOwnProperty.call`, we ensure that the `hasOwnProperty` method is correctly scoped and avoid any potential issues with pollution in the environment.

**Comment by 549915217** [comment]
Describe the bug
Currently, this problem has been discovered in the Qiankun environment

**Comment by josdejong** [comment]
Thanks for reporting. I've fixed this via df4ac5abf3ea92a4cdaab16e6ea8dfa7f5ccb726 (not yet published).

On a side note: the `lruQueue` function can use some extra unit testing I think: all tests pass when I remove the two occurrences of this line of code:

```js
while (!Object.prototype.hasOwnProperty.call(queue, ++base)) {}
```

If anyone can help improve the tests please let me know.

**Comment by 549915217** [comment]
I hope to have a new version updated as soon as possible. Thank you.
